### PR TITLE
RK-16077 - Add Tutorial for Containerized Go Deployment

### PR DESCRIPTION
**Pull request by Urook**
Add Tutorial for Containerized Go Deployment

### CODE REVIEW

File: docs/go-container-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+---
+id: go-container-tutorial
+title: "Deploy Rookout on a Go container"
+sidebar_label: Go Container
+---
+
+---
+
+*Deploying the Golang SDK requires name/password to our private repository.*
```
Comment (LiranLast) [inline docs/go-container-tutorial.md:9]: Why so scary? Let's create a much smaller CTA below :)

File: docs/go-container-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+```
+
+Configuration is where you can get fancy. You have got additional options up your sleeve:
+1. Move options from environment variables to secret or configuration managers.
+2. If you are using a [Rookout Controller](etl-controller-intro), set up the remote host and port configuration.
+3. Dig deeper into other options available right [here](go-setup#start).
+
+### 3. Embed the Full GoSDK During Deployment
+After obtaining the credentials for our private repository, follow these instructions:
+1. Get the ARTIFACTORY_CREDS build-argument in your Dockerfile:
```
Comment (LiranLast) [inline docs/go-container-tutorial.md:72]: Let's combine into a single step? No sub-steps? :)

File: docs/go-container-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+```bash
+go mod tidy
+```
+
+Configuration is where you can get fancy. You have got additional options up your sleeve:
+1. Move options from environment variables to secret or configuration managers.
+2. If you are using a [Rookout Controller](etl-controller-intro), set up the remote host and port configuration.
+3. Dig deeper into other options available right [here](go-setup#start).
+
+### 3. Embed the Full GoSDK During Deployment
```
Comment (LiranLast) [inline docs/go-container-tutorial.md:70]: There are five sub-steps here - looks a bit awkward :)

Let's try to reduce the number of steps by making them optional or merging them.

On the other hand, it's ok to have more top-level steps :)

File: docs/go-container-tutorial.md (hunk 1 above)
Comment (Urook) [reply docs/go-container-tutorial.md:9]: No preblem. BTW it is the same message as the one in our go sdk setup tutorial

File: docs/go-container-tutorial.md (hunk 3 above)
Comment (Urook) [reply docs/go-container-tutorial.md:70]: Thing is, a lot of these come with examples for each os. How do you think we should go about combining them?

File: docs/go-container-tutorial.md (hunk 1 above)
Comment (LiranLast) [reply docs/go-container-tutorial.md:9]: We'll fix that one later :)

File: docs/go-container-tutorial.md (hunk 3 above)
Comment (LiranLast) [reply docs/go-container-tutorial.md:70]: We use these snippets for per OS instructions 
https://docs.rookout.com/docs/python-setup/#dependencies

File: docs/go-container-tutorial.md (hunk 1 above)
Comment (Urook) [reply docs/go-container-tutorial.md]: > Why so scary? Let's create a much smaller CTA below :)

Is it better now?

File: docs/go-container-tutorial.md (hunk 3 above)
Comment (Urook) [reply docs/go-container-tutorial.md:62]: > We use these snippets for per OS instructions https://docs.rookout.com/docs/python-setup/#dependencies

Done

File: docs/go-container-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+Start by:
+```bash
+git clone https://github.com/Rookout/go-tutorial-2022
+cd go-tutorial-2022
+```
+
+### 1. Add the stub GoSDK Package
+
+Rookout for Go is a private package.
```
Comment (LiranLast) [inline docs/go-container-tutorial.md:22]: This requires some copy work - I'll take a look later

File: docs/go-container-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+2. If you are using a [Rookout Controller](etl-controller-intro), set up the remote host and port configuration.
+3. Dig deeper into other options available right [here](go-setup#start).
+
+### 3. Embed the Full GoSDK During Deployment
+After obtaining the credentials for our private repository, follow these instructions:
+1. Get the ARTIFACTORY_CREDS build-argument in your Dockerfile:
+```docker
+ARG ARTIFACTORY_CREDS
+```
+2. Set the following Go environment variables in your Dockerfile before building the project
```
Comment (LiranLast) [inline docs/go-container-tutorial.md:68]: No internal numbering within steps :)

Either have top-level steps, or trim it down otherwise...

**Comment by Urook** [comment]
@hopemanryan BTW note that this PR needs to be updated, approced and merged

File: docs/go-container-tutorial.mdx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+As your Go application spins up, search for this output at the top:
+<img src="/img/screenshots/go_success.png" />
+
+Interact with your application at `http://localhost:8080` and use Rookout to debug it on the fly!
+
+### Questions?
+
+1. Check out this reference [implementation](https://github.com/Rookout/go-tutorial-2022/compare/configure-rookout).
```
Comment (Urook) [inline docs/go-container-tutorial.mdx:140]: This tutorial is old, we should redo the configure-rookout branch

File: docs/go-container-tutorial.mdx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+Start by:
+```bash
+git clone https://github.com/Rookout/go-tutorial-2022
+cd go-tutorial-2022
+```
+
+### 1. Add the GoSDK Package
+
+Start out by adding The Rookout package:
```
Comment (LiranLast) [inline docs/go-container-tutorial.mdx:22]: Start repeats itself as the first word :)

File: docs/go-container-tutorial.mdx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+```bash
+
+go get -d github.com/Rookout/GoSDK
+
+```
+
+### 2. Start Rookout
+
+Load and start the package to connect to your Rookout account (if you haven't signed up, do that [here](https://app.rookout.com/#mode=signUp)).
```
Comment (LiranLast) [inline docs/go-container-tutorial.mdx:32]: There's no load in Go, maybe import?

File: docs/go-container-tutorial.mdx [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+		Token: "[Your Rookout Token]",
+		Labels: map[string]string{"env": "dev"},
+	})
+    // ...
+}
+```
+
+<div class="rookout-org-info"></div>
+
+It's always best to run `tidy` after adding a new dependency:
```
Comment (LiranLast) [inline docs/go-container-tutorial.mdx:55]: I think we can skip this section

File: docs/go-container-tutorial.mdx [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+```bash
+
+go mod tidy
+
+```
+
+Make sure you install additional dependencies [if necessary](go-setup#dependencies).
```
Comment (LiranLast) [inline docs/go-container-tutorial.mdx:63]: Let's move that to the build stage? (3)

**Comment by LiranLast** [comment]
Looks good to me. Do we want to merge the example before we merge this?

**Comment by naomirookout** [comment]
@LiranLast already updated - https://github.com/Rookout/go-tutorial-2022/compare/configure-rookout

**Comment by LiranLast** [comment]
Amazing!

**Review by LiranLast — APPROVED** [review]

**Pull request merged**